lib/addons/velero/index.ts

```diff
@@ -0,0 +1,259 @@
+                config:{}
+            },
+        },
+        serviceAccount: {
+            server:{}
+        }
+    },
+
+};
+
+export class VeleroAddOn implements ClusterAddOn {
+
+    private options: VeleroAddOnProps;
+    private chartNode: HelmChart;
+    constructor(props?: VeleroAddOnProps) {
+        if (props) {
+            // merge the nested json files
+            this.options = merge(defaultProps, props);
+        }
+        else {
+            this.options = defaultProps
+        }
+    }
+
+    /**
+     * Implementation of the add-on contract deploy method.
+    */
+    async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        const props = this.options;
+               
+        // Create S3 bucket if no existing bucket, create s3 bucket and corresponding KMS key
+        const bucketName = this.returnS3BucketName(clusterInfo, "backup-bucket", props.values.configuration.backupStorageLocation.bucket)
+
+        // Create Namespace if namespace is not explicied defined.
+        const veleroNamespace = this.createNamespaceIfNeeded(clusterInfo, "velero", props.namespace);
+
+        // Setup IAM Role for Service Accounts (IRSA) for the Velero Service Account
+        const veleroServiceAccountName = this.createIRSA(clusterInfo, "velero-account", veleroNamespace, bucketName)
+        
+        // Setup the values for the helm chart
+        const valueVariable: VeleroAddOnProps = {
+            values: {
+                configuration: {
+                    backupStorageLocation: {
+                        prefix: props.values.configuration.backupStorageLocation.prefix ?? "velero/" + cluster.clusterName,
+                        bucket: bucketName,
+                        config:{
+                           region: props.values.configuration.backupStorageLocation.config.region ?? cluster.stack.region,
+                        }
+                    },
+                    volumeSnapshotLocation:{
+                        config:{
+                            region: props.values.configuration.backupStorageLocation.config.region ?? cluster.stack.region
+                        }
+                    }
+                },
+                // IAM role for Service Account
+                serviceAccount: {
+                    server: {
+                        create: false,
+                        name: veleroServiceAccountName,    
+                    }
+                }             
+            }
+        };
+
+        const values = merge(props.values, valueVariable.values) ?? {}; 
+ 
+        this.chartNode = cluster.addHelmChart("velero-addon", {
+            chart: "velero",
+            repository: "https://vmware-tanzu.github.io/helm-charts/",
+            release: Constants.SSP_ADDON,
+            namespace: veleroNamespace,
+            version: props.version,
+            values: values
+        });
+        return this.chartNode;
```

Comment:
You need to add dependency on the service account before returning here.

Reply:
resolved.